core/payment-driver/dummy/src/service.rs

```diff
@@ -117,8 +119,9 @@ async fn schedule_payment(
     };
 
     // Spawned because calling payment service while handling a call from payment service
-    // would result in a deadlock.
+    // would result in a deadlock. We need to wait a bit, so parent scope be able to answer
     Arbiter::spawn(async move {
+        std::thread::sleep(Duration::from_millis(100));
```

Comment:
small nit: would not import this at the top of the file or only 1 use
```suggestion
        std::thread::sleep(actix::clock::Duration::from_millis(100));
```
```suggestion
        std::thread::sleep(Duration::from_millis(100));
```